### PR TITLE
Add Pydocstyle Checking for All Function Docstrings

### PR DESCRIPTION
**Pull request by gkapfham**
This pull request adds support for `pydocstyle` to check the docstrings inside of the Python source code.

### What is the current behavior?

Currently, the docstrings are not checked for consistency.

### What is the new behavior if this PR is merged?

The docstrings are in a uniform format and are passing the `pydocstyle` checks.

##### This PR has:

- [ ] Commit messages that are correctly formatted
- [ ] Tests for newly introduced code
- [ ] Docstrings for newly introduced code

This PR is a small change that fixes the fact that the docstrings are inconsistent.

#### Developers

@gkapfham

### CODE REVIEW

**Comment by gkapfham** [comment]
Hi @Michionlion and @schultzh and @corlettim, I am currently working on this PR and hope to finish it very soon. Currently, `pydocstyle` is flagging a lot of small errors, but I should be able to easily fix them! Let me know if you notice any mistakes as I am working. Thanks!

**Comment by schultzh** [comment]
Is ```pydocstyle``` a new file that checks errors within our code? Or is it integrated in existing files?

**Comment by gkapfham** [comment]
Hi @schultzh thanks for your question. We have decided to adopt Pydocstyle so that we ensure that all of our docstrings are formatted in a consistent fashion. If you check the `scripts/lint.sh` file you can see the way in which we are running the tool. If you want to run it for a specific file you could type, for instance, a command like: `pipenv run pydocstyle gator/arguments.py`. I will keep working on this PR today, with plans to improve all of the comments today and have this ready for merging!

**Comment by gkapfham** [comment]
Hi @corlettim and @schultzh and @Michionlion I have finished fixing all of the docstring problems. The entire GatorGrader codebase now adheres to the Pydocstyle standard. I also made some minor improvements to some of the comments in the code base.

File: tests/test_arguments.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def test_valid_argument_combinations_accepted(chosen_arguments):
     ],
 )
 def test_invalid_argument_combinations_accepted(chosen_arguments):
-    """Check that not valid argument combinations do not verify correctly"""
+    """Check that not valid argument combinations do not verify correctly.."""
```
Comment (Michionlion) [inline tests/test_arguments.py]: There are some duplicate periods probably introduced by find-and-replace. It can probably be fixed with more!

Comment (gkapfham) [reply tests/test_arguments.py]: Thanks for noticing this mistake, I have fixed it @Michionlion.

**Comment by gkapfham** [comment]
Hi @Michionlion and @schultzh and @corlettim, I have fixed the noted mistakes in the docstrings and I judge that the code is ready for you to review it again. Thanks!

**Pull request merged**